apps/src/client.rs

```diff
@@ -227,50 +251,57 @@ pub fn connect(
                 break 'read;
             }
 
-            let (len, from) = match socket.recv_from(&mut buf) {
-                Ok(v) => v,
-
-                Err(e) => {
-                    // There are no more UDP packets to read, so end the read
-                    // loop.
-                    if e.kind() == std::io::ErrorKind::WouldBlock {
-                        trace!("recv() would block");
-                        break 'read;
-                    }
-
-                    return Err(ClientError::Other(format!(
-                        "recv() failed: {:?}",
-                        e
-                    )));
-                },
-            };
+            for event in &events {
+                let token: usize = event.token().into();
+                let socket = &sockets[token];
+                let (len, from) = match socket.recv_from(&mut buf) {
+                    Ok(v) => v,
+
+                    Err(e) => {
+                        // There are no more UDP packets to read, so end the read
+                        // loop.
+                        if e.kind() == std::io::ErrorKind::WouldBlock {
+                            trace!("recv() would block");
```

Comment:
we probably want to add detail to this line to say which socket would block

Reply:
Indeed. I added in the log the local address on which the socket is bound.